server/server.go

```diff
@@ -79,6 +81,10 @@ func (s *Server) Run(ctx context.Context, addr string, port int, tlsCert string,
 		if err != nil {
 			return fmt.Errorf("error creating unix socket listener: %w", err)
 		}
+		err = updateSocketPermission(socketPath, conf.Server.UnixSocketPerm)
+		if err != nil {
+			return fmt.Errorf("error updating permission of unix socket file: %w", err)
+		}
```

Comment:
For better readability, it would be better to move this whole block to a function that returns the listener and an error.